aspnetcore/blazor/components/cascading-values-and-parameters.md

```diff
@@ -94,6 +94,192 @@ builder.Services.AddCascadingValue(sp =>
 
+Calling <xref:Microsoft.AspNetCore.Components.CascadingValueSource%601.NotifyChangedAsync%2A> to issue update notifications can be used to signal multiple Razor component subscribers that a cascading value has changed. Notifications aren't possible for subscribers that adopt static server-side rendering (static SSR), so subscribers must adopt an interactive render mode. 
+
+In the following example:
+
+* `NotifyingDalek` implements <xref:System.ComponentModel.INotifyPropertyChanged> to notify clients that a property value has changed. When the `Units` property is set, the <xref:System.ComponentModel.PropertyChangedEventHandler> (`PropertyChanged`) is invoked.
+* The `SetUnitsToOneThousand` method can be triggered by subscribers to set `Units` to 1,000 with a simulated processing delay.
+
+Keep in mind for production code that any change in state (any property value change of the class) causes all subscribed components to rerender, regardless of which part of the state they use. We recommend creating granular classes, cascading them separately with specific subscriptions to ensure that only components subscribed to a specific portion of the application state are affected by changes.
+
+`NotifyingDalek.cs`:
+
+```csharp
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+
+public class NotifyingDalek : INotifyPropertyChanged
+{
+    public event PropertyChangedEventHandler? PropertyChanged;
+    private int units;
+
+    public int Units
+    {
+        get => units;
+        set
+        {
+            if (units != value)
+            {
+                units = value;
+                OnPropertyChanged();
+            }
+        }
+    }
+
+    protected virtual void OnPropertyChanged(
+        [CallerMemberName] string? propertyName = default)
+            => PropertyChanged?.Invoke(this, new(propertyName));
+
+    public async Task SetUnitsToOneThousand()
+    {
+        // Simulate a three second delay in processing
+        await Task.Delay(3000);
+
+        Units = 1000;
+    }
+}
+```
+
+The following `CascadingStateServiceCollectionExtensions` creates a <xref:Microsoft.AspNetCore.Components.CascadingValueSource%601> from a type that implements <xref:System.ComponentModel.INotifyPropertyChanged>.
+
+`CascadingStateServiceCollectionExtensions.cs`:
+
+```csharp
+using System.ComponentModel;
+using Microsoft.AspNetCore.Components;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+public static class CascadingStateServiceCollectionExtensions
+{
+    public static IServiceCollection AddCascadingStateNotifier<T>(
```

Comment:
I kind of don't like the `AddCascadingStateNotifier` name. It suggests that we're registering something that should have `Notifier` in its type name or interface, which it doesn't.  

Maybe something like:  
- `AddNotifyingCascadingValue()`  
- `AddNotifyPropertyChangedCascadingValue()`  

The issue isn't with the definition but with how it's used:  

```csharp
builder.Services.AddCascadingStateNotifier<NotifyingDalek>(new NotifyingDalek() { Units = 888 });
```  

The registration should instead look something like this, leveraging generic type inference:  

```csharp
builder.Services.AddNotifyingCascadingValue(new NotifyingDalek() { Units = 888 });
```
...we are actually registering a _cascading value_, which should be clear.

Reply:
Ok ... I'll fix this separately after all of the other changes go in.

Reply:
... and I'm mopping up the last few changes locally that I missed with the suggestion commit.